File: src/com/company/classes/Coordinates.java
package com.company.classes;

import java.util.regex.PatternSyntaxException;

public class Coordinates implements com.company.superclass.Coordinates {
    private int x;
    private double y;

    public Coordinates() {}

    public Coordinates(String fromString) {
        new Coordinates(fromString, false);
    }

    public Coordinates(String fromString, boolean csv) {
        String[] arguments;
        try {
            arguments = fromString.split(csv? ", ": " " , 2);
        } catch (PatternSyntaxException | NullPointerException e) {
            throw new IllegalArgumentException("Error in argument: " + e.getMessage() + ".");
        }
        try {
            setX(Integer.parseInt(arguments[0]));
            setY(Double.parseDouble(arguments[1]));
        } catch (NumberFormatException | NullPointerException | ArrayIndexOutOfBoundsException e) {
            throw new IllegalArgumentException("Can't parse Coordinates from \"" + fromString + "\": " + e.getMessage() + ".");
        }
    }

    public Coordinates(int x, double y) throws IllegalArgumentException {
        setX(x);
        setY(y);
    }

    @Override
    public int getX() {
        return x;
    }

    @Override
    public void setX(int x) {
        if (x > -497) {
            this.x = x;
        }
        else {
            throw new IllegalArgumentException("Out of range (min -497)");
        }
    }

    @Override
    public double getY() {
        return y;
    }

    @Override
    public void setY(double y) {
        if (y < 268.0) {
            this.y = y;
        }
        else {
            throw new IllegalArgumentException("Out of range (max 268.0)");
        }
    }

    @Override
    public String getCoordinates(Boolean csv) {
        return getX() + (csv? ", ": " ") + getY();
    }
}

File: src/com/company/classes/Coordinates.kt
package com.company.classes

import java.util.regex.PatternSyntaxException
import kotlin.NumberFormatException
import kotlin.properties.Delegates

class Coordinates : com.company.superclasses.Coordinates {

    private var x by Delegates.notNull<Int>()
    private var y by Delegates.notNull<Double>()


    constructor(fromString: String) {
        val arguments: Array<String>
        try {
            arguments = fromString.split(" ", limit = 2).toTypedArray()
        } catch (e: PatternSyntaxException) {
            throw IllegalArgumentException("Error in argument: " + e.message + ".")
        }
        try {
            if (arguments.size > 1) {
               if (arguments[0].toInt() > -497 && arguments[1].toDouble() < 268.0) {
                   this.x = arguments[0].toInt()
                   this.y = arguments[1].toDouble()
               }
            }
        } catch (e: NumberFormatException) {
            throw IllegalArgumentException("Can't parse Coordinates from \"" + fromString + "\": " + e.message + ".")
        }
    }

    constructor(x: Int, y: Double) {
        this.x = x
        this.y = y
    }

    override fun getCoordinates(csv: Boolean): String {
        var string: String = ""
        try {
            string = if (csv) {
                "$x, $y"
            } else {
                "$x $y"
            }
        } catch (e: IllegalStateException) {
            throw IllegalArgumentException("Can't get coordinates, because this don't init.")
        }
        return string

    }

}